src/WaveTrack.cpp

```diff
@@ -2442,12 +2432,20 @@ void WaveTrack::UpdateLocationsCache() const
       const WaveClip *prev = nullptr;
       for (const auto clip : clips)
       {
+         //enough for estimation
          num += clip->NumCutLines();
 
-         if (prev && fabs(prev->GetEndTime() -
-                          clip->GetStartTime()) < WAVETRACK_MERGE_POINT_TOLERANCE)
+         if (prev && fabs(prev->GetPlayEndTime() -
+                          clip->GetPlayStartTime()) < WAVETRACK_MERGE_POINT_TOLERANCE)
             ++num;
 
+         /*auto& cutlines = clip->GetCutLines();
+         for (auto& cut : cutlines)
+         {
+             auto cutlinePosition = clip->GetSequenceStartTime() + cut->GetSequenceStartTime();
+             if (clip->WithinPlayRegion(cutlinePosition))
+                 ++num;
```

Comment:
Some doubtful code you dont need now?

Reply:
Yes, but now I think it's not that important to precisely count only visible cutlines here, since it's used only as an estimation for cache preallocation size and assertion at the end.  Removing it